ls1intum/Artemis
# `Exam mode`: Fix breadcrumbs when assessing exam exercises

### Checklist
#### General
#### Client

### Motivation and Context
During the exam mode testing we noticed that opening the exercise assessment dashboard loses the exam context. This means that navigating back into the exam using the breadcrumbs no longer works. You instead get into the course assessment dashboard, which looks similar but does not contain the exam exercises, which can be confusing. 

### Description
We now keep the examId in the urls while navigating to the assessment dashboard and when starting an assessment.

While the components support this already, some of the necessary routes were missing.

### Steps for Testing
Prerequisites:
- 1 Tutor
- 1 Exam with exercises of each type

1. Open the exercise assessment dashboard as a tutor (exams -> assessment -> exercise dashboard)
2. From now on, check the breadcrumbs with every button press and make sure the exam context gets maintained.
3. Start new assessments, open existing ones, etc. (do typical assessment tasks).

### Review Progress

#### Performance Review
#### Code Review
#### Manual Tests
